### PR TITLE
add title editing functionality

### PR DESCRIPTION
**Pull request by colinfruit**

### CODE REVIEW

File: src/components/Content/RightPanel/TaskDetails/TaskTitle/TaskTitle.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  toggleInput = () => {
+    const { showInput } = this.state;
+    this.setState({showInput: !showInput});
+  }
+
+  onEnter = (value) => {
+    const { id, patchTask } = this.props;
+    // const { newTitle } = this.state;
```
Comment (elimisteve) [inline src/components/Content/RightPanel/TaskDetails/TaskTitle/TaskTitle.js]: @colinfruit Need this comment?

File: src/components/Content/RightPanel/TaskDetails/TaskTitle/TaskTitle.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    patchTask({
+      gid: id,
+      title: value
+    })
+  }
+
+  onChange = (e) => {
+    const { value } = e.target;
+    const { key } = e;
```
Comment (elimisteve) [inline src/components/Content/RightPanel/TaskDetails/TaskTitle/TaskTitle.js]: @colinfruit Could do `const { key, target: { value } = e;`, FYI :-) but this is fine :+1:

**Comment by elimisteve** [comment]
@colinfruit Looks great! Well done :+1: about to pull and double-check, but the code is simple and clean.

**Comment by colinfruit** [comment]
thanks, I can add these two improvements if you want. 👍

**Comment by elimisteve** [comment]
Sure that'd be great

**Comment by elimisteve** [comment]
@colinfruit Also: if `key === 'Escape'`, stop editing and reset `newTitle` to ''

**Comment by colinfruit** [comment]
Sure

**Comment by elimisteve** [comment]
(I know I didn't mention that before, but I just realized that letting users stop editing would be useful. Could even set `showInput: false` onBlur... not sure.)

**Comment by colinfruit** [comment]
Yes, no problem.

**Comment by colinfruit** [comment]
For onBlur() let's just return to original?

**Comment by elimisteve** [comment]
Yup, perfect.

File: src/components/Content/RightPanel/TaskDetails/TaskTitle/TaskTitle.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+
+import './TaskTitle.css';
+
+class TaskTitle extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newTitle: '',
```
Comment (elimisteve) [inline src/components/Content/RightPanel/TaskDetails/TaskTitle/TaskTitle.js]: @colinfruit Can delete this line, yes?

Comment (colinfruit) [reply src/components/Content/RightPanel/TaskDetails/TaskTitle/TaskTitle.js]: already done

**Comment by colinfruit** [comment]
Changes committed, thanks a lot for the help

**Comment by elimisteve** [comment]
Merged! Great work, @colinfruit :tada: :tada: :tada:

**Pull request merged**